iceoryx2/src/service/attribute.rs

```diff
@@ -99,59 +99,99 @@
 
 use core::ops::Deref;
 
-use iceoryx2_bb_container::{byte_string::FixedSizeByteString, vec::FixedSizeVec};
+use iceoryx2_bb_container::{semantic_string::SemanticString, vec::FixedSizeVec};
 use iceoryx2_bb_derive_macros::ZeroCopySend;
 use iceoryx2_bb_elementary::CallbackProgression;
 use iceoryx2_bb_log::fatal_panic;
 use serde::{Deserialize, Serialize};
 
-use crate::constants::{MAX_ATTRIBUTES, MAX_ATTRIBUTE_KEY_LENGTH, MAX_ATTRIBUTE_VALUE_LENGTH};
+use crate::constants::MAX_ATTRIBUTES;
+
+/// Module containing the key type used for service attributes.
+pub mod key {
+
+    use core::hash::Hash;
+    use core::hash::Hasher;
+
+    use iceoryx2_bb_container::semantic_string;
+    use iceoryx2_bb_container::semantic_string::SemanticString;
+    use iceoryx2_bb_derive_macros::ZeroCopySend;
+
+    use crate::constants::MAX_ATTRIBUTE_KEY_LENGTH;
+
+    semantic_string! {
+      /// Fixed string for service attribute keys.
+      name: FixedString,
```

Comment:
Since it is a string with a semantic meaning and content it should not have a generic name. Can we remove the `key` namespace and call it `AttributeKey` and the other one `AttributeValue`. All the other semantic strings are named in a similar fashion (see. `GroupName`, `UserName` or `FilePath` for instance).

Reply:
With this it would also fit better to the C++ binding where they are called `Attribute::Key` and `Attribute::Value` (they are aliases inside a class and now thinking it would have made sense to name them `AttributeKey` and `AttributeValue`) Something I will adjust!

Reply:
Made aliases:

```rust
type AttributeKey = key::FixedString;
type AttributeValue = key::FixedString;
```

The modules are required due to (perhaps?) a bug in `semantic_string!` meaning it can only be used once per module.